### ISSUE TITLE
resource: log reason for "allocate cannot find resources for $jobid"

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/flux-sched#409]
Even for simple jobspecs, it can be sometimes difficult to determine exactly why resources weren't found.  One example that bit me recently was the jobspec included a `rack` in the request, but when the `resource` module was instatiated with `hwloc`, no `rack` resources existed at all in the graph database.

Providing meaningful messages for every possible scenario is unrealistic, but I think there is some low hanging fruit we can start with.  For example, if pruning happens at the root resource, then just logging which child resource caused the prune should be sufficient. Another example, for resource types that just don't exist at all in the resource graph, we should be able to provide a meaningful message.

**Comment by dongahn** [comment]
Yup, this makes a lot of sense. Since resource uses recursive algorithms, how we will bubble up the reasons for an unmatch to the upper layer requires some thoughts. Let's discuss what would be best.

**Comment by dongahn** [comment]
Maybe we can have some log print statements when traverser return `-1` from either depth first walk.

**Comment by dongahn** [comment]
Do we need more logging than the current method? We do log whether a job cannot be scheduled because of satisfiability or internal bug or just a lack of resources at the time of scheduling.

**Comment by SteVwonder** [comment]
> We do log whether a job cannot be scheduled because of satisfiability or internal bug or just a lack of resources at the time of scheduling.

:+1: for logging when the job is complete unsatisfiable.  That is a huge improvement IMO!

It might be useful to differentiate between unsatisfiable because you asked for a resource that doesn't exist at all in the graph vs. asking for too much of a resource that does exist.  The former "should be relatively easy":tm: to check for since we have a `by_type` map in the db already.  I think this is relatively low priority and can be lumped into any "hardening" sprints we have.
